File: Practica3/Marco teorico/6. DIP/DIP.py
class IBanco:
    def procesarPago(self, monto):
        pass

class Banco(IBanco):
    def procesarPago(self, monto):
        print("Pago procesado en el banco por $" + str(monto))

class ProcesadorDePagos:
    def __init__(self, banco):
        self.banco = banco

    def procesarPago(self, monto):
        self.banco.procesarPago(monto)
        print("Pago procesado exitosamente")


banco = Banco()
procesador = ProcesadorDePagos(banco)
procesador.procesarPago(100.0)

File: Practica3/Marco teorico/6. DIP/DIP.java
public interface IBanco {
    void procesarPago(double monto);
}

public class Banco implements IBanco {
    @Override
    public void procesarPago(double monto) {
        System.out.println("Pago procesado en el banco por $" + monto);
    }
}

public class ProcesadorDePagos {
    private IBanco banco;

    public ProcesadorDePagos(IBanco banco) {
        this.banco = banco;
    }

    public void procesarPago(double monto) {
        banco.procesarPago(monto);
        System.out.println("Pago procesado exitosamente");
    }
}

public class Main {
    public static void main(String[] args) {
        IBanco banco = new Banco();
        ProcesadorDePagos procesador = new ProcesadorDePagos(banco);
        procesador.procesarPago(100.0);
    }
}